src/kani-compiler/cbmc/src/goto_program/expr.rs

```diff
@@ -1205,7 +1203,18 @@ impl Expr {
     /// `ArithmeticOverflowResult r; >>>r.overflowed = builtin_add_overflow(self, e, &r.result)<<<`
     pub fn add_overflow(self, e: Expr) -> ArithmeticOverflowResult {
         let result = self.clone().plus(e.clone());
-        let overflowed = self.add_overflow_p(e);
+        // FIXME: https://github.com/model-checking/kani/issues/786
+        // Overflow checking on pointers is hard to do at the IREP level.
+        // Instead, we rely on `--pointer-overflow-check` in CBMC.
+        let overflowed = if self.typ.is_pointer() {
```

Comment:
Don't we have to check if `e` is a pointer here?

Reply:
Technically, that we'd notice any cases cause type-checking would fail. But making the test is clearer.